.github/workflows/ci.yaml

```diff
@@ -78,10 +78,17 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
+        with:
+          path: main
+      - name: Fetch slurmctld
+        uses: actions/checkout@v4
+        with:
+          repository: charmed-hpc/slurmctld-operator
+          path: slurmctld-operator
       - name: Setup operator environment
         uses: charmed-kubernetes/actions-operator@main
         with:
           provider: lxd
-          juju-channel: 3.1/stable
+          juju-channel: 3.4/stable
       - name: Run tests
-        run: tox run -e integration -- --charm-base=${{ matrix.bases }}
+        run: cd main && tox run -e integration -- --charm-base=${{ matrix.bases }} --use-local
```

Comment:
[nit]: I typically like to use `pushd` instead of `cd` when I need to change directories in a script. That way if we ever need to change back to the root directory to switch into another directory, all you have to do is call `popd`. No need to track the old pwd with `$OLDPWD` or a shell variable.

If you still want to use `cd` though, I'm cool with that.

Reply:
Yeah, let's just keep it simple for now. Though, we should switch to `pushd` if we add more steps in the future.

Reply:
Agreed then :+1: